### ISSUE TITLE
add test coverage for FoodComposition

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/sd#532]
https://github.com/GroceriStar/sd/tree/master/src/data/FoodComposition

**Comment by atherdon** [comment]
@AndrewSerra

**Comment by AndrewSerra** [comment]
This is the same as all tests I assume, right? Doing this now.

**Comment by AndrewSerra** [comment]
Again sorry about this week being a bit away because of the hospital issues I've been busy with getting health tests done.

**Comment by AndrewSerra** [comment]
for the test coverage (#532) I read and saw the json files being in generator/projects/FoodComposition/ directory. Will the json files stay in that directory or will they be moved to src/data/FoodComposition/? I'm asking this because files.js uses the directory src/data/FoodComposition/

Another question is should I add them to the files.js file to use them for the test? if yes, should they be separate for countries or one test?

**Comment by atherdon** [comment]
@vadim9999

**Comment by vadim9999** [comment]
@AndrewSerra 
> for the test coverage (#532) I read and saw the json files being in generator/projects/FoodComposition/ directory. Will the json files stay in that directory or will they be moved to src/data/FoodComposition/? I'm asking this because files.js uses the directory src/data/FoodComposition/
> 
> Another question is should I add them to the files.js file to use them for the test? if yes, should they be separate for countries or use just one test?

jsons files will be moved from generator/projects/FoodComposition to src/data/FoodComposition, but now.
Yes, add separated for countries and FoodComposition.json to the files.js. Use separated by countries and FoodComposition.json for tests.

**Comment by AndrewSerra** [comment]
awesome working on it

**Comment by AndrewSerra** [comment]
as a suggestion, do you think having a single test file (foodComposition.test.js) containing all tests for the countries and foodComposition.json could be better? Since all tests are in a way "related" to FoodComposition I wanted to ask whether the test file can cover them all. or should I keep the test files all separate files as well?

**Comment by atherdon** [comment]
for now it's debatable because after finishing this tests we something like closing a door for this sub-project. But if Vadim told so - then I'm voting for his way

**Comment by AndrewSerra** [comment]
ok I'll keep all files separate and simple, no problem

**Comment by atherdon** [comment]
yeah, maybe it will save some time for people that will work or use this
part in the future.
That set is a pretty unique combination of data that our machine learning
team created. When it will be released to public - i'm curious to see a
response

**Comment by AndrewSerra** [comment]
it sounds like something awesome is happening :) I saw the comments in [this issue](https://github.com/GroceriStar/sd/issues/387) interesting.

> yeah, maybe it will save some time for people that will work or use this part in the future. That set is a pretty unique combination of data that our machine learning team created. When it will be released to public - i'm curious to see a response

**Comment by atherdon** [comment]
this is just a tip of the iceberg :)
I go insane few months ago and decide to go fully open about this project.
and I don't have internet for a month. and I'm very active at my work. I
mean I cannot just relax and switch my mind away. so about a month I wrote
a lot of notes and combine a lot of information that I collect for the last
years about food tech, recipes, and all these things.
if you interested - there a huge longread at Hackernoon forum:
https://community.hackernoon.com/t/food-alliance-project-diary/2034/

>
